packages/react/src/analytics/integrations/Guardian/providers/Riskified/Riskified.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  EventData,
+  integrations,
+  LoadIntegrationEventData,
+  StrippedDownAnalytics,
+  TrackTypes,
+  TrackTypesValues,
+} from '@farfetch/blackout-analytics';
+import get from 'lodash/get';
+import type { RiskifiedIntegrationOptions } from './types';
+
+class Riskified extends integrations.Integration<RiskifiedIntegrationOptions> {
+  isReady: boolean;
+  siteId: string;
+  baseUrl: string;
+
+  /**
+   * Creates an instance of Riskified and sets the script on the page.
+   *
+   * @param options - Custom options for the integration.
+   * @param loadData - Analytics's load event data.
+   * @param analytics - Stripped down analytics instance.
+   */
+  constructor(
+    options: RiskifiedIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, analytics);
+
+    this.isReady = false;
+    this.siteId = options.siteId || 'farfetch.com';
+    this.baseUrl = `https://beacon.riskified.com?shop=${this.siteId}&sid=`;
+
+    this.setupScript(loadData);
+  }
+
+  /**
+   * Returns true due to being a required integration - No need to check for consent.
+   *
+   * @returns A value indicating if the integration should load.
+   */
+  static override shouldLoad(): boolean {
+    return true;
+  }
+
+  /**
+   * Creates the script based on the correlation `id` and appends it on the page.
+   * Once the script is on the page, it will automatically call `RISKX.go()`.
+   *
+   * @param loadData - Analytics's load event data.
+   *
+   * @returns The Riskified instance to allow chaining calls.
+   */
+  setupScript(loadData: LoadIntegrationEventData): this {
+    const correlationId = get(loadData, 'user.localId');
+    const script = document.createElement('script');
+    const url = `${this.baseUrl}${correlationId}`;
+
+    document.body.appendChild(script);
+
+    script.onload = this.onLoad;
+
+    script.setAttribute('data-test', 'riskified');
+    script.async = true;
+    script.src = url;
+
+    return this;
+  }
+
+  /**
+   * Tracks events (page views or page actions).
+   *
+   * @param data - The event object.
+   *
+   */
+  override track(data: EventData<TrackTypesValues>) {
+    switch (data.type) {
+      case TrackTypes.PAGE:
+        this.trackPage(get(data, 'context.web.window.location.href'));
+    }
+  }
+
+  /**
+   * Tracks the page view event. Calls the global Riskified variable RISKX.go with the web location.
+   *
+   * @param location - The location data.
+   *
+   */
+  trackPage(location: string) {
+    if (!this.isReady) {
+      return;
```

Comment:
Related with https://github.com/Farfetch/blackout/pull/649/files#r1097218215, since it is returning here when it is not ready, when it is ready, it should call `RISKX.go` (which would make the description for the `setupScript` correct) since the event will be lost. I am not sure if instead of just returning, the current event data should be added to a queue, which would then be emptied when the script was loaded. Please, take a look at it since the way it is now, some page tracks can be lost.

Reply:
I've added a promise to track the page load to ensure that events are captured properly. This way, we can guarantee that the events will only run after the script has been fully loaded, and none event will be lost.

Reply:
Thanks!